openlibrary/plugins/upstream/account.py

```diff
@@ -551,16 +552,31 @@ def GET(self):
 class account_validation(delegate.page):
     path = '/account/validate'
 
+    def ia_username_exists(username):
+        url = "https://archive.org/metadata/@%s" % username
+        try:
+            res = requests.get(url).json()
+            if not res:
+                return True
+            return False
+        except (OSError, ValueError):
+            return
+
     @staticmethod
-    def validate_username(username):
+    def validate_username(self, username):
         if not 3 <= len(username) <= 20:
             return _('Username must be between 3-20 characters')
         if not re.match('^[A-Za-z0-9-_]{3,20}$', username):
             return _('Username may only contain numbers and letters')
+
         ol_account = OpenLibraryAccount.get(username=username)
         if ol_account:
             return _("Username unavailable")
 
+        ia_account = self.ia_username_exists(username)
```

Comment:
Should be able to do something like this and drop the `self`  / switch to `staticmethod`

```suggestion
        ia_account = account_validation.ia_username_exists(username)
```